### PR TITLE
Dotnet worker: Allow binding to BlobContainerClient without blob

### PR DESCRIPTION
**Pull request by aishwaryabh**
### Issue describing the changes in this PR

resolves https://github.com/Azure/azure-functions-dotnet-worker/issues/1337

For binding `BlobContainerClient` as an input binding, we are required to provide blob name in the result such as `BlobInput("input-container/blob-name")`. We should be allowed to do this without providing the blob name, and currently we run into an error if you try to just put down the container name:
`Invalid blob path specified : 'input-container'. Blob identifiers must be in the format 'container/blob'`

I added a change to the webjobs extension which loosens the parsing and validating checks by adding a new parameter to `BlobPath.ParseAndValidate` called `isParameterBindingData`. This value is only set to true if we are binding to a parameter binding data type and then skips some of the checks which caused this to fail initially.

This PR adds a check in the dotnet worker to ensure that blob name cannot be null if we are not binding to a `BlobContainerClient`, which is the check we are skipping on the webjobs extension.

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
  * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

File: samples/WorkerBindingSamples/Blob/BlobInputBindingSamples.cs [hunk 1]
```diff
@@ -114,5 +114,20 @@ public BlobInputBindingSamples(ILogger<BlobInputBindingSamples> logger)
 
             return req.CreateResponse(HttpStatusCode.OK);
         }
+
+        [Function(nameof(BlobContainerClientInputFunction))]
+        public HttpResponseData BlobContainerClientInputFunction(
```
Comment (liliankasem) [inline samples/WorkerBindingSamples/Blob/BlobInputBindingSamples.cs:119]: nit: move to be above the BlobClient sample
nit2: rename to `BlobInputContainerClientFunction` to match the naming pattern of the rest of the functions

Comment (aishwaryabh) [reply samples/WorkerBindingSamples/Blob/BlobInputBindingSamples.cs:119]: Changed!

File: .gitignore [hunk 2]
```diff
@@ -4,6 +4,7 @@
 ## Get latest from https://github.com/github/gitignore/blob/master/VisualStudio.gitignore
 
 Azure.Functions.Cli
+azurite
```
Comment (liliankasem) [inline .gitignore:7]: This is odd, it already exists in the feature branch

https://github.com/Azure/azure-functions-dotnet-worker/blob/80a2659320b7b20cf9697a2b6a9cda75e780185c/.gitignore#L7

Comment (liliankasem) [reply .gitignore:7]: AH, you need to rebase. I updated feature/sdk-binding to be caught up with main

**Review by liliankasem — APPROVED** [review]

File: extensions/Worker.Extensions.Storage.Blobs/src/BlobStorageConverter.cs [hunk 3]
```diff
@@ -160,6 +160,11 @@ private bool IsBlobExtension(ModelBindingData bindingData)
                 throw new ArgumentNullException(nameof(containerName));
             }
 
+            if (targetType != typeof(BlobContainerClient) && string.IsNullOrEmpty(blobName))
```
Comment (surgupta-msft) [inline extensions/Worker.Extensions.Storage.Blobs/src/BlobStorageConverter.cs:163]: @liliankasem Highlighting that this check may change when we remove `CollectionModelBindingData`. Below code is valid and it does not have target type as `BlobContainerClient`

`BlobInput("container-path") IEnumerable<string> blobs`

**Review by surgupta-msft — APPROVED** [review]

File: extensions/Worker.Extensions.Storage.Blobs/src/BlobStorageConverter.cs (hunk 3 above)
Comment (liliankasem) [reply extensions/Worker.Extensions.Storage.Blobs/src/BlobStorageConverter.cs:163]: Ah yup I have that covered in my [PR](https://github.com/Azure/azure-functions-dotnet-worker/pull/1690)

**Comment by liliankasem** [comment]
Going to abandon this as it will be covered in another PR: https://github.com/Azure/azure-functions-dotnet-worker/pull/1690

**Pull request closed without merging**